src/State/Setup/Threads.hs

```diff
@@ -63,6 +67,23 @@ startUserRefreshThread lock session requestChan = void $ forkIO $ forever refres
               Right upd -> return upd
           threadDelay (seconds userRefreshInterval)
 
+-- This thread refreshes the map of typing users every second, forever,
+-- to expire users who have stopped typing. Expiry time is 3 seconds.
+startTypingUsersRefreshThread :: RequestChan -> IO ()
+startTypingUsersRefreshThread requestChan = void $ forkIO $ forever refresh
+  where
+    seconds = (* (1000 * 1000))
```

Comment:
This needs to be much less frequent - refreshing every second causes a screen redraw every second, so this could probably be okay at, say, 5 seconds.

Reply:
setting the expiry interval to 5 seconds (as the web client does) and the refresh interval to 3 seconds (half the expiry interval)